libraries/lib-audio-io/AudioIO.cpp

```diff
@@ -2015,104 +2023,208 @@ bool AudioIO::ProcessPlaybackSlices(
          // warping
          if (frames > 0) {
             size_t produced = 0;
+            // Check for asynchronous user changes in mute, solo status
+            const auto discardable = SequenceShouldBeSilent(*mPlaybackSequences[iSequence]);
             if (toProduce)
                produced = mixer->Process(toProduce);
+
             //wxASSERT(produced <= toProduce);
             // Copy (non-interleaved) mixer outputs to one or more ring buffers
-            const auto nChannels = mPlaybackSequences[iSequence++]->NChannels();
+            const auto nChannels = mPlaybackSequences[iSequence]->NChannels();
+            
             for (size_t j = 0; j < nChannels; ++j) {
-               auto warpedSamples = mixer->GetBuffer(j);
-               const auto put = mPlaybackBuffers[iBuffer++]->Put(
-                  warpedSamples, floatSample, produced, frames - produced);
-               // wxASSERT(put == frames);
-               // but we can't assert in this thread
-               wxUnusedVar(put);
+
+               auto& buffer = mProcessingBuffers[iBuffer++];
+               const auto offset = buffer.size();
+               //there could be leftovers from the previous pass, don't discard them
+               buffer.resize(buffer.size() + frames, 0);
```

Comment:
I would add a comment that this `resize` is expected not to allocate memory in the worker thread, because sufficient size should have been reserved in `AllocateBuffers`.  Perhaps, test capacity before and after and `assert` no change.